extensions/positron-r/src/provider.ts

```diff
@@ -362,63 +460,47 @@ async function findRBinaryFromPATHNotWindows(whichR: string): Promise<string | u
-	if (!userPath) {
-		// If we didn't find R in the default user location, check WOW64
-		userPath = await getRegistryInstallPath('HKEY_CURRENT_USER', 'WOW6432Node');
+	if (os.platform() !== 'win32') {
+		LOGGER.info('Skipping registry check on non-Windows platform');
+		return undefined;
 	}
-	let machinePath = await getRegistryInstallPath('HKEY_LOCAL_MACHINE');
-	if (!machinePath) {
-		// If we didn't find R in the default machine location, check WOW64
-		machinePath = await getRegistryInstallPath('HKEY_LOCAL_MACHINE', 'WOW6432Node');
+
+	// eslint-disable-next-line @typescript-eslint/naming-convention
+	const Registry = await import('@vscode/windows-registry');
+
+	const hives: any[] = ['HKEY_CURRENT_USER', 'HKEY_LOCAL_MACHINE'];
+	const wows = ['', 'WOW6432Node'];
+
+	let installPath = undefined;
+
+	for (const hive of hives) {
+		for (const wow of wows) {
+			const R64_KEY: string = `SOFTWARE\\${wow ? wow + '\\' : ''}R-core\\R64`;
+			try {
+				const key = Registry.GetStringRegKey(hive, R64_KEY, 'InstallPath');
+				if (key) {
+					installPath = key;
+					LOGGER.info(`Registry key ${hive}\\${R64_KEY}\\InstallPath reports an R installation at ${key}`);
+					break;
+				}
+			} catch { }
+		}
 	}
-	if (!userPath && !machinePath) {
+
+	if (installPath === undefined) {
+		LOGGER.info('Cannot determine current version of R from the registry.');
 		return undefined;
 	}
-	const installPath = userPath || machinePath || '';
 
 	const binPath = firstExisting(installPath, binFragments());
 	if (!binPath) {
 		return undefined;
 	}
-	LOGGER.info(`Identified the current version of R from the registry: ${binPath}`);
+	LOGGER.info(`Identified the current R binary: ${binPath}`);
 
 	return binPath;
 }
 
-/**
- * Get the registry install path for R.
- *
- * @param hive The Windows registry hive to check -- HKCU or HKLM
- * @param wow Optionally, the WOW node to check under `Software`. R's install
- * path is written to a WOW (Windows on Windows) node when e.g. an x86 build of
- * R is installed on an ARM version of Windows.
- *
- * @returns The install path for R, or undefined if an R installation file could
- * not be found at the install path.
- */
-async function getRegistryInstallPath(hive: 'HKEY_CURRENT_USER' | 'HKEY_LOCAL_MACHINE', wow?: string | undefined): Promise<string | undefined> {
-	// 'R64' here is another place where we explicitly ignore 32-bit R
-	// Amend a WOW path after "Software" if requested
-	const R64_KEY: string = `SOFTWARE\\${wow ? wow + '\\' : ''}R-core\\R64`;
-
-	if (os.platform() !== 'win32') {
-		LOGGER.info('Skipping registry check on non-Windows platform');
-		return undefined;
-	}
-
-	// eslint-disable-next-line @typescript-eslint/naming-convention
-	const Registry = await import('@vscode/windows-registry');
-
-	try {
-		LOGGER.info(`Checking for 'InstallPath' in registry key ${hive}\\${R64_KEY}`);
-		return Registry.GetStringRegKey(hive, R64_KEY, 'InstallPath');
-	} catch (err) {
-		LOGGER.info(err as string);
```

Comment:
This caused lines like this to appear in the output for positron-r:

```
2024-09-25 17:21:29.514 [info] Checking for 'InstallPath' in registry key HKEY_CURRENT_USER\SOFTWARE\R-core\R64
2024-09-25 17:21:29.515 [info] Unable to open registry key
```

and folks often wondered if this was a newsworthy error, possibly explaining some problem they were having. The answer to that is almost always "no". So the inlined registry checks are now silent if the key does not exist to eliminate this red herring.